### PR TITLE
Update selected date in state when prop changes

### PR DESCRIPTION
**Pull request by jesuscc1993**
The selected prop (which is a date and not a boolean, contrary to what its name may suggest) defines the date that is selected in the agenda.

Currently, however, this prop is only read in the constructor. If the prop later changes the selected date in the agenda state does not update. This PR fixes that.

I had to check that selected has a value as the component would crash if I saved an undefined selected date in the state.

### CODE REVIEW

**Comment by Inbal-Tish** [comment]
@jesuscc1993 Hi. Sorry for the late response. Can you please resolve conflicts so I can merge?

**Comment by jesuscc1993** [comment]
@Inbal-Tish conflicts resolved.

File: src/agenda/index.js [hunk 1]
```diff
@@ -139,13 +138,21 @@ export default class AgendaView extends Component {
...
+    }
+    if (
+      nextProps.selected &&
+      !dateutils.sameDate(nextProps.selected, this.props.selected)
+    ) {
+      nextState.selectedDay = parseDate(nextProps.selected);
+    }
+    if (Object.keys(nextState).length) {
+      this.setState(nextState);
     } else {
```
Comment (Inbal-Tish) [inline src/agenda/index.js:154]: I think the 'else' statement is misplaced. It should be attached to the `if(props.items)`, no?

**Comment by emilisb** [comment]
Hi @jesuscc1993 

Any plans to finish with this?

**Review by jesuscc1993 — COMMENTED** [review]
> Hi @jesuscc1993
> 
> Any plans to finish with this?

I'm confused, what is still pending on my side?

File: src/agenda/index.js (hunk 1 above)
Comment (Inbal-Tish) [reply src/agenda/index.js:154]: Yes, you're right

**Comment by Inbal-Tish** [comment]
@jesuscc1993 Can you please resolve conflicts so we can merge?

**Comment by Inbal-Tish** [comment]
@jesuscc1993 Hi. You need to merge master and fix lint errors.

**Comment by jesuscc1993** [comment]
> @jesuscc1993 Hi. You need to merge master and fix lint errors.

I'm working on the lint issues; I'm having trouble getting the command to work in my local.
I'll ping you when I'm done.

File: src/agenda/index.js [hunk 2]
```diff
@@ -92,7 +92,9 @@ export default class AgendaView extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (!prevProps.items) {
+    if (this.props.selected && !dateutils.sameDate(parseDate(this.props.selected), parseDate(prevProps.selected))) {
+      this.setState({selectedDay: parseDate(this.props.selected)});
```
Comment (Inbal-Tish) [inline src/agenda/index.js:96]: Are 'this.currentMonth' and 'topDay' shouldn't be updated as well?

Comment (jesuscc1993) [reply src/agenda/index.js:96]: We don't use those props in out project so we don't know them.
I wouldn't feel confident on doing anything with them as I don't trust I'd be able to properly test their changes.

**Comment by jesuscc1993** [comment]
@Inbal-Tish @emilisb conflicts solved, lint issues fixed and PR tested
ping me if you need anything else

**Comment by jguix** [comment]
Please review this PR, it is very important for our project to stay updated to the main branch and keep these changes.

File: src/agenda/index.js (hunk 2 above)
Comment (jguix) [reply src/agenda/index.js:96]: This is a simple pull request, and it is actually fixing a bug. Could we please move this forward so we can have this issue fixed in the main branch? I understand that reviewing and testing PRs takes time, but we are also investing time as collaborators fixing it, and the issue has been open for a long time :)

If I were to review it, I would say everything is ok now.

**Review by jguix — APPROVED** [review]

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**